### DISCUSSION TITLE
Riverpod generator StateNotifier initial method

### DISCUSSION THREAD

**Discussion by NijatTagizada** [rrousselGit/riverpod#2320] [Q&A]
Hi. How can I write this provider with riverpod_generator 

```dart
final termsNotifierProvider =
    StateNotifierProvider.autoDispose<TermsNotifier, TermsState>((ref) {
  return TermsNotifier(ref.watch(termsRepositoryProvider))
    ..fetchTerms();
});
```
I want to call ..fetchTerms() method when provider initialized

fetchTerms is void method

**Comment by AhmedLSayed9** [comment]
I guess you're manually handling loading/error states which is a bit legacy and riverpod can do that for you now.

If you just want to fetch some terms asynchronously, use FutureProvider (async function if you use code-gen).
If you want to fetch some terms asynchronously and be able to modify them by some side-effect methods, use AsyncNotifierProvider (class with async build if you use code-gen)

**Comment by scopendo** [comment]
Invoke it from the build method of `TermsNotifier`.

See https://github.com/rrousselGit/riverpod/discussions/2253#discussioncomment-5196249 for a very basic example of a notifier class based on Riverpod generator syntax.
